src/palette/internal/palette.cpp

```diff
@@ -515,102 +508,143 @@ Palette::Type Palette::guessType() const
     case ElementType::ACTION_ICON: {
         const ActionIcon* action = toActionIcon(e);
         QString actionCode = QString::fromStdString(action->actionCode());
 
         if (actionCode.contains("beam")) {
-            return Type::Beam;
+            return Palette::Type::Beam;
         }
         if (actionCode.contains("grace") || actionCode.contains("acciaccatura") || actionCode.contains("appoggiatura")) {
-            return Type::GraceNote;
+            return Palette::Type::GraceNote;
         }
         if (actionCode.contains("frame") || actionCode.contains("box") || actionCode.contains("measure")) {
-            return Type::Layout;
+            return Palette::Type::Layout;
         }
-        return Type::Custom;
+        return Palette::Type::Custom;
     }
     default: {
         if (e->isSpanner()) {
-            return Type::Line;
+            return Palette::Type::Line;
         }
         if (e->isTextBase()) {
-            return Type::Text;
+            return Palette::Type::Text;
         }
     }
     }
 
-    return Type::Custom;
+    return Palette::Type::Custom;
 }
 
-std::function<void(PaletteCellPtr)> Palette::cellHandlerByPaletteType(const Palette::Type& type) const
+void scaleCell(PaletteCellPtr cellPtr)
 {
-    switch (type) {
-    case Type::Bracket:
-        return [](PaletteCellPtr cellPtr) {
-            if (!cellPtr || !cellPtr->element || !cellPtr->element.get()->isBracket()) {
-                return;
-            }
-
-            Bracket* bracket = toBracket(cellPtr->element.get());
+    if (!cellPtr || !cellPtr->element) {
+        return;
+    }
+
+    Palette::Type guessedType = guessTypeFrom(cellPtr->element.get());
+    cellPtr->mag = appropriateMagnificationFor(guessedType);
+
+    // Some special cases
+    if (cellPtr->element.get()->isLayoutBreak()) {
+        cellPtr->mag = 1.2;
+    } else if (cellPtr->element.get()->isSpacer()) {
+        cellPtr->mag = 0.7;
+    } else if (cellPtr->element.get()->isBracket()) {
+        Bracket* bracket = toBracket(cellPtr->element.get());
+        if (bracket->bracketType() == BracketType::BRACE) {
+            bracket->setStaffSpan(0, 1);
+            cellPtr->mag = 1.2 * appropriateMagnificationFor(Palette::Type::Bracket);
+        }
+    }
+}
 
-            if (bracket->bracketType() == BracketType::BRACE) {
-                bracket->setStaffSpan(0, 1);
-                cellPtr->mag = 1.2;
-            }
-        };
+qreal appropriateMagnificationFor(const Palette::Type& type)
+{
+    switch (type) {
+    case Palette::Type::Bracket:
+        return 0.6;
+    case Palette::Type::KeySig:
+    case Palette::Type::BarLine:
+    case Palette::Type::Clef:
+    case Palette::Type::BagpipeEmbellishment:
+    case Palette::Type::Line:
+    case Palette::Type::TimeSig:
+        return 0.8;
+    case Palette::Type::Repeat:
+    case Palette::Type::Text:
+        return 0.85;
+    case Palette::Type::Tempo:
+        return 0.9;
```

Comment:
Previously, different cells in the "tempo" palette had different magnifications (some were ≈0.98, some were ≈0.85). A magnification of 0.9 is a compromise between the previous values.